ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/views/HiveAggregateInsertDeleteIncrementalRewritingRule.java

```diff
@@ -116,7 +121,7 @@ protected IncrementalComputePlanWithDeletedRows createJoinRightInput(RelOptRuleC
     aggInput = HiveHepExtractRelNodeRule.execute(aggInput);
     aggInput = new HiveRowIsDeletedPropagator(relBuilder).propagate(aggInput);
 
-    int rowIsDeletedIdx = aggInput.getRowType().getFieldCount() - 1;
+    int rowIsDeletedIdx = aggInput.getRowType().getFieldCount() - 2;
```

Comment:
should we add some comment for this magic number `-2`?

Reply:
This was removed by #5053

Reply:
Added comment.